core-java-modules/core-java-string-conversions-2/src/main/java/com/baeldung/stringtocamelcase/StringtocamelcaseApplication.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.stringtocamelcase;
+
+import com.google.common.base.CaseFormat;
+import com.ibm.icu.lang.UCharacter;
+import com.ibm.icu.text.BreakIterator;
+import org.apache.commons.text.CaseUtils;
+import org.apache.commons.text.WordUtils;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class StringtocamelcaseApplication {
+
+	public static void main(String[] args) {
+
+	}
+
+
+	//Core Java Solutions
+	public static String toCamelCaseByIteration(String text, char delimiter) {
+		if (text == null || text.isEmpty()) return text;
+		boolean shouldConvertNextCharToLower = false;
+		StringBuilder builder = new StringBuilder();
+
+		for (int i = 0; i < text.length(); i++) {
+			char currentChar = text.charAt(i);
+
+			if (currentChar == delimiter) {
+				shouldConvertNextCharToLower = false;
+			} else if (shouldConvertNextCharToLower) {
+				builder.append(Character.toLowerCase(currentChar));
+			} else {
+				//Bingo!!! We,ve seen a Word Breaker
+				//This also caters for converting the first character to Uppercase
+				builder.append(Character.toUpperCase(currentChar));
+				shouldConvertNextCharToLower = true;
+			}
+
+		}
+
+		return builder.toString();
+	}
+
+	public static String toCamelCaseBySplitting(String text, String delimiter) {
+		if (text == null || text.isEmpty()) return text;
+		String [] words = text.split(delimiter);
+
+		StringBuilder builder = new StringBuilder();
+
+		for (String word : words) {
+			//Convert the first character to Uppercase and others to lowercase
+			// e.g sTRING =====> String
+			word = word.isEmpty() ? word : Character.toUpperCase(word.charAt(0)) + word.substring(1).toLowerCase();
```

Comment:
Would this work with input "A_A_A_A_A_A" ? which should result in "AAAAAA"

Reply:
Yes this would work